src/litserve/specs/openai.py

```diff
@@ -258,12 +279,12 @@ def _encode_response(self, output: Union[Dict[str, str], List[Dict[str, str]]])
             )
             logger.exception(error)
             raise HTTPException(500, error)
-
-        return ChatMessage(**message)
+        usage_info = self.extract_usage_info(message)
+        return {**message, **usage_info}
 
     def encode_response(
         self, output_generator: Union[Dict[str, str], List[Dict[str, str]]], context_kwargs: Optional[dict] = None
-    ) -> ChatMessage:
+    ) -> Generator[Dict, None, None]:
```

Comment:
This is making the spec less clear, let's create a proper type (doesn't ChatMessage have optional usage info already?)

Reply:
usage info is set in the outermost level of ChatCompletionResponse object. If we yield a ChatMessage object then it  is converted into a dict (`{"role":..., "content":...}`) when it flows to the main process through the multiprocessing connection Pipe. 

Would you recommend to do something like - 

```
EncodeGenerator = Generator[Union[Dict, ChatMessage], None, None]

def encode_response(
        self, output_generator: Union[Dict[str, str], List[Dict[str, str]]], context_kwargs: Optional[dict] = None
    ) -> EncodeGenerator:
```

Reply:
This still feels too complex. Can't we add an optional usage section to `ChatMessage`? Or even better
```
Generator[Union[ChatMessage, ChatMessageWithUsage]]
```

where we define `ChatMessageWithUsage` explicitly?

Reply:
yes, we can. one question - when we provide the type hint as `ChatMessage` or `ChatMessageWithUsage` which are Pydantic classes, we also allow users to yield dictionaries. Will this type hint create confusion?

Reply:
This is only for `encode_response` right? So most users won't override it, but at least they will know what flows where by glancing at the code.

Reply:
make sense! thank you for clarifying.

Reply:
updated to `Iterator[Union[ChatMessage, ChatMessageWithUsage]]` since Generator has a verbose type hint `Generator[yield_type, send_type, return_type]`